assets/src/edit-story/components/canvas/singleSelectionMovable.js

```diff
@@ -170,20 +180,22 @@ function SingleSelectionMovable( {
 				frame.resize = [ newWidth, newHeight ];
 				frame.translate = drag.beforeTranslate;
 				if ( shouldAdjustFontSize ) {
-					target.style.fontSize = calculateFitTextFontSize( target.firstChild, height, width );
+					target.style.fontSize = calculateFitTextFontSize( target.firstChild, height, width, minMaxFontSize );
 				}
 				setTransformStyle( target );
 			} }
 			onResizeEnd={ ( { target } ) => {
-				if ( frame.resize[ 0 ] !== 0 && frame.resize[ 1 ] !== 0 ) {
+				const editorWidth = frame.resize[ 0 ];
+				const editorHeight = frame.resize[ 1 ];
+				if ( editorWidth !== 0 && editorHeight !== 0 ) {
 					const properties = {
-						width: frame.resize[ 0 ],
-						height: frame.resize[ 1 ],
-						x: selectedElement.x + frame.translate[ 0 ],
-						y: selectedElement.y + frame.translate[ 1 ],
+						width: editorToDataX( editorWidth ),
+						height: editorToDataY( editorHeight ),
+						x: selectedElement.x + editorToDataX( frame.translate[ 0 ] ),
+						y: selectedElement.y + editorToDataY( frame.translate[ 1 ] ),
```

Comment:
I'd probably move the two translate values to variables above for readability:

```js
const [ deltaX, deltaY ] = frame.translate;
```

Reply:
Done